util/chunk/column.go

```diff
@@ -244,19 +244,32 @@ const (
 )
 
 // resize resizes the column so that it contains n elements, only valid for fixed-length types.
-func (c *Column) resize(n, typeSize int) {
+func (c *Column) resize(n, typeSize int, isNull bool) {
 	sizeData := n * typeSize
 	if cap(c.data) >= sizeData {
 		(*reflect.SliceHeader)(unsafe.Pointer(&c.data)).Len = sizeData
 	} else {
 		c.data = make([]byte, sizeData)
 	}
 
+	newNulls := false
 	sizeNulls := (n + 7) >> 3
 	if cap(c.nullBitmap) >= sizeNulls {
 		(*reflect.SliceHeader)(unsafe.Pointer(&c.nullBitmap)).Len = sizeNulls
 	} else {
 		c.nullBitmap = make([]byte, sizeNulls)
+		newNulls = true
+	}
+	if isNull && newNulls {
+		// do nothing, all slots are already set to null
+	} else {
```

Comment:
why not use `!isNull || !newNulls`?

Reply:
`isNull && newNulls` is clearer than `!isNull || !newNulls`?

Reply:
I think it's ok to use `!isNull || !newNulls`

Reply:
OK, it's updated to `isNull || !newNulls` now~